docs/internet-identity-spec.adoc

```diff
@@ -125,12 +126,14 @@ where `msg` is a value with of type
    kind: "authorize-client";
    sessionPublicKey: Uint8Array;
    maxTimeToLive?: bigint;
+   derivationOrigin?: string;
  }
 
 where
 
 * the `sessionPublicKey` contains the public key of the session key pair.
 * the `maxTimeToLive`, if present, indicates the desired time span (in nanoseconds) until the requested delegation should expire. The Identity Provider frontend is free to set an earlier expiry time, but should not create a larger.
+* the `derivationOrigin`, if present, indicates an origin that should be used for principal derivation instead of the client origin. Values must match the following regular expression: `^https:\/\/[\w-]*(\.raw)?\.ic0\.app$`. Internet Identity will only accept values that are also listed in the https://internetcomputer.org/docs/current/references/ic-interface-spec/#state-tree-canister-information[_public_ custom WASM section] `ii_alternative_domains` of the corresponding canister (see <<alternative-frontend-origins>>).
```

Comment:
Shouldn’t this be alternative _origins_, rather than URLs? I expect most applications want to require `https` here. And probably not just any port, but only specific ones.

Reply:
Isn't the pattern implicitly describing the origin `(https, [\w-]*(\.raw)?\.ic0\.app, 443)`?

Reply:
Yes, I am referring to the name. But below ii_alternative_origins is used, so maybe just a typo.

Reply:
@nomeata: from the perspective of the canister, there can be multiple alternative origins, thus the custom section (or well-known URL) will be called `alternative_origins`. But this message is sent by the front-end hosted on one of the `alternative_origins` requesting a derivation using the (single) original one, so `derivationOrigin` seems appropriate for me?

Reply:
Yes, that’s all appropriate! I am merely commenting on the `ii_alternative_domains` in the diff :-)

Reply:
Ah, I missed that. Will update in a bit :-)  Thanks